katana/native/cases/static/cases/js/casesApp.js

```diff
@@ -0,0 +1,1040 @@
+	 return; 
+	}
+	var oldCaseSteps = caseApp.jsonCaseObject["Steps"]['step'];
+	var newCaseSteps = new Array(listCases.length);
+		
+	for (xi=0; xi < listCases.length; xi++) {
+		var xtr = listCases[xi];
+		var ni  = xtr.getAttribute("data-sid");
+		console.log(xi + " => " + ni);
+		newCaseSteps[ni] = oldCaseSteps[xi];
+	}
+
+	caseApp.jsonCaseObject["Steps"]['step'] = newCaseSteps;
+	caseApp.jsonCaseSteps  = caseApp.jsonCaseObject["Steps"]
+	caseApp.mapCaseJsonToUi(caseApp.jsonCaseSteps);
+		
+	},
+
+// Removes a test suite by its ID and refresh the page. 
+	removeTestStep: function ( sid ){
+		caseApp.jsonCaseSteps['step'].splice(sid,1);
+		console.log("Removing testcases "+sid+" now " + Object.keys(caseApp.jsonCaseSteps).length);
+		caseApp.mapCaseJsonToUi(caseApp.jsonCaseSteps);
+	},
+
+// Create a fresh step at the end of the table.
+	addNewTestStepToUI: function() {
+	var newObj = caseApp.createNewStep();
+
+	if (!caseApp.jsonCaseSteps['step']) {
+		caseApp.jsonCaseSteps['step'] = [];
+		}
+	if (!jQuery.isArray(caseApp.jsonCaseSteps['step'])) {
+		caseApp.jsonCaseSteps['step'] = [caseApp.jsonCaseSteps['step']];
+		}
+
+	console.log("Adding new step", caseApp.jsonCaseSteps,caseApp.jsonCaseSteps['step']);
+	caseApp.jsonCaseSteps['step'].push(newObj);  // Don't delete anything
+	caseApp.mapCaseJsonToUi(caseApp.jsonCaseSteps);		
+	},
+
+// Inserts a new test step 
+	addTestStepAboveToUI: function (sid,xdata,copy) {
+		var newObj = caseApp.createNewStep();
+		var aid = 0;   // Insert here. 
+		if (sid < 1) { 
+			aid = 0 ;
+		} else {
+			aid = sid;                // One below the current one. 
+		}
+
+		if (!caseApp.jsonCaseSteps['step']) {
+			caseApp.jsonCaseSteps['step'] = [];
+			}
+		if (!jQuery.isArray(caseApp.jsonCaseSteps['step'])) {
+			caseApp.jsonCaseSteps['step'] = [caseApp.jsonCaseSteps['step']];
+			}
+
+		if (copy == 1){
+			console.log("Copying..., ", sid, " from ", caseApp.jsonCaseSteps['step'][sid]);
+			newObj = jQuery.extend(true, {}, caseApp.jsonCaseSteps['step'][sid]); 
+			}
+		caseApp.jsonCaseSteps['step'].splice(aid,0,newObj);  // Don't delete anything
+		caseApp.mapCaseJsonToUi(caseApp.jsonCaseSteps);		
+		},
+
+	// for a popup 
+	redrawArguments: function(sid, oneCaseStep,popup) {
+		var arguments = oneCaseStep['Arguments']['argument'];
+		caseApp.makePopupArguments(popup, oneCaseStep);
+	},
+
+
+	saveOneArgument: function( sid, aid, xdata) {
+		var obj = caseApp.jsonCaseSteps['step'][sid]['Arguments']['argument'][aid]; 	
+		obj['@name'] = katana.$activeTab.find('[argid=caseArgName-'+aid+']').attr('value');
+		obj['@value'] = katana.$activeTab.find('[argid=caseArgValue-'+aid+']').attr('value');
+		console.log("Saving..arguments-div "+ sid + " aid = "+ aid);
+		console.log(katana.$activeTab.find('[argid=caseArgValue-'+aid+']'));
+		console.log(katana.$activeTab.find('[argid=caseArgValue-'+aid+']'));
+		console.log(obj);
+	//mapTestStepToUI(sid, xdata);
+	},
+
+ 	addOneArgument: function( sid , xdata, popup ) {
```

Comment:
Make addOneArgument and insertOneArgument one function

Reply:
Ok